### PR TITLE
create restaurant mongoose schema and model

### LINKED ISSUE #2: create restaurant mongoose schema and model

### PR DESCRIPTION
**Pull request by JanaAbuHaltam**
The model is designed to capture essential information about a restaurant and is equipped with necessary validations.
Further enhancements, such as specifying the data type for restaurantPicture  and setting default values, can be considered in future updates.
closes #2

### CODE REVIEW

File: server/server/Models/restaurant.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    type: [String],
+    required: true,
+  },
+
+  phoneNumber: {
+    type: String,
+    required: true,
+  },
+
+  restaurantPicture: {
```
Comment (belalninja) [inline server/server/Models/restaurant.js:22]: this field will be string which represents the path to image file inside the backend

File: server/server/Models/restaurant.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    // type to be decieded later on
+    // type: String,
+    required: true,
```
Comment (belalninja) [inline server/server/Models/restaurant.js:25]: we can put a placeholder image in frontend if restaurant image is absent from database

**Review by belalninja — CHANGES_REQUESTED** [review]
well done, check requested changes

File: server/server/Models/restaurant.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+// schema
+const restaurantSchema = new mongoose.Schema({
+  _id: mongoose.Schema.Types.ObjectId,
+
+  name: {
+    type: String,
+    required: true,
+  },
+
+  location: {
```
Comment (belalninja) [inline server/server/Models/restaurant.js:12]: location will be based on longitude and latitude, so it should be an array of only 2 numbers

File: server/server/Models/restaurant.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    // default
+    // default img to be decided
+  },
+
+  acceptedPayment: {
+    type: [String],
+    required: true,
+  },
+
+  category: {
```
Comment (belalninja) [inline server/server/Models/restaurant.js:35]: category can have multiple values, so it should be an array of strings

**Review by belalninja — CHANGES_REQUESTED** [review]
i have requested other changes, please check them

File: server/server/Models/restaurant.js (hunk 1 above)
Comment (JanaAbuHaltam) [reply server/server/Models/restaurant.js:22]: makes sense, i'll edit it

**Review by belalninja — APPROVED** [review]
well done jana

**Pull request merged**